apps/dashboard/src/main/java/com/akto/utils/RiskScoreOfCollections.java

```diff
@@ -0,0 +1,213 @@
+import com.akto.dto.type.SingleTypeInfo;
+import com.mongodb.BasicDBObject;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.model.Aggregates;
+import com.mongodb.client.model.BulkWriteOptions;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Projections;
+import com.mongodb.client.model.UpdateManyModel;
+import com.mongodb.client.model.UpdateOptions;
+import com.mongodb.client.model.Updates;
+import com.mongodb.client.model.WriteModel;
+
+public class RiskScoreOfCollections {
+
+    private List<ApiInfoKey> getUpdatedApiInfos(int timeStampFilter){
+        List<ApiInfoKey> updatedApiInfoKeys = new ArrayList<>();
+
+        // get freshly updated testing run issues here
+        Bson projections = Projections.include("_id", TestingRunIssues.LAST_SEEN);
+        Bson filters = Filters.gte(TestingRunIssues.LAST_SEEN, timeStampFilter);
+
+        List<TestingRunIssues> issues = new ArrayList<>();
+        try {
+            issues = TestingRunIssuesDao.instance.findAll(filters, projections);   
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        // after getting issues, get updated apiinfokeys related to that issues only
+        if(issues == null || issues.size() == 0){
+            return updatedApiInfoKeys;
+        }
+        for(TestingRunIssues issue: issues){
+            TestingIssuesId issueId = issue.getId();
+            ApiInfoKey apiInfoKey = issueId.getApiInfoKey();
+            updatedApiInfoKeys.add(apiInfoKey);
+        }
+
+        return updatedApiInfoKeys;
+    }
+
+    private Map<ApiInfoKey, Float> getSeverityScoreMap(List<TestingRunIssues> issues){
+        // Method to calculate severity Score for the apiInfo on the basis of HIGH, LOW, MEDIUM
+        InventoryAction inventoryAction = new InventoryAction();
+        Map<ApiInfoKey, Float> severityScoreMap = new HashMap<>();
+
+        for(TestingRunIssues issue: issues){
+            String severity = issue.getSeverity().toString();
+            Float severityScore = (Float) inventoryAction.calculateRiskValueForSeverity(severity);
+            ApiInfoKey apiInfoKey = issue.getId().getApiInfoKey();
+            if(severityScoreMap.isEmpty() || !severityScoreMap.containsKey(apiInfoKey)){
+                severityScoreMap.put(apiInfoKey, severityScore);
+            }else{
+                Float prev = severityScoreMap.get(apiInfoKey);
+                severityScore += prev;
+                severityScoreMap.put(apiInfoKey, severityScore);
+            }
+        }
+
+        return severityScoreMap;
+    }
+
+    public void updateSeverityScoreInApiInfo(int timeStampFilter){
+        ArrayList<WriteModel<ApiInfo>> bulkUpdatesForApiInfo = new ArrayList<>();
+        List<ApiInfoKey> updatedApiInfoKeys = getUpdatedApiInfos(timeStampFilter) ;
+
+        if(updatedApiInfoKeys == null || updatedApiInfoKeys.size() == 0){
+            return ;
+        }
+        Bson filterQ = Filters.and(
```

Comment:
add a log in case in case there are some updatedApiInfoKeys

Reply:
done